core/src/main/java/org/apache/accumulo/core/spi/compaction/ExecutorManager.java

```diff
@@ -31,4 +31,9 @@
    * Create a thread pool executor within a compaction service.
    */
   public CompactionExecutorId createExecutor(String name, int threads);
+
+  /**
+   * @return an id for a configured external execution queue.
+   */
+  public CompactionExecutorId getExternalExecutor(String name);
```

Comment:
The interface was not there so that users could implement their own, I was just using it to logically group things that went together to avoid having lots of unrelated groups of methods interleaved in a single interface.